tokio-quiche/src/quic/io/worker.rs

```diff
@@ -458,6 +450,34 @@ where
         buffer_write_outcome
     }
 
+    /// Selects a network path, if none already selected.
+    ///
+    /// This will return the first path available in the write state's
+    /// `pending_paths` iterator. If that is empty a new iterator will be
+    /// created by querying quiche itself.
+    ///
+    /// Note that the connection's statically configured local address will be
+    /// used to query quiche for available paths, so this can't handle multiple
+    /// local addresses currently.
+    fn select_path(
+        &mut self, qconn: &QuicheConnection,
+    ) -> Option<(SocketAddr, SocketAddr)> {
+        if self.write_state.selected_path.is_some() {
+            return self.write_state.selected_path;
+        }
+
+        let from = self.cfg.local_addr;
+
+        // Initialize paths iterator.
+        if self.write_state.pending_paths.len() == 0 {
+            self.write_state.pending_paths = qconn.paths_iter(from);
+        }
+
+        let to = self.write_state.pending_paths.next()?;
+
+        Some((from, to))
```

Comment:
Can you add a `debug_assert!(self.write_state.selected_path.is_none())` to catch future mistakes.